.github/scripts/check_native_dispatch.py

```
@@ -0,0 +1,103 @@
+
+    caps = list(PRIORITY)
+    probe = "#include \"zbuild.h\"\n#include \"arch_functions.h\"\n" \
+            "#define S2(x) #x\n#define S(x) S2(x)\n"
+    for cp in caps:
+        probe += f"#ifdef {cp}\n#pragma message(\"DISP {cp}=\" S({cp}))\n#endif\n"
+    for nm in NATIVE_MACROS:
+        probe += f"#ifdef {nm}\n#pragma message(\"NAT {nm}\")\n#endif\n"
+
+    with tempfile.NamedTemporaryFile("w", suffix=".c", delete=False) as f:
+        f.write(probe); tu = f.name
+    p = subprocess.run([cc, "-fsyntax-only", *defs, *extra,
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)